app/src/util/ingest_utils.py

```diff
@@ -58,3 +59,18 @@ def process_and_ingest_sys_args(argv: list[str], logger: Logger, ingestion_call:
         db_session.commit()
 
     logger.info("Finished processing")
+
+
+def tokenize(text: str) -> list[str]:
+    """
+    The add_special_tokens argument is specified in PreTrainedTokenizerFast.encode_plus(), parent class of MPNetTokenizerFast.
+    It defaults to True for encode_plus() but defaults to False for .tokenize().
+    Setting add_special_tokens=True will add the special tokens CLS(0) and SEP(2) to the beginning and end of the input text.
+    """
+    tokenizer = app_config.sentence_transformer.tokenizer
+    if isinstance(tokenizer, PreTrainedTokenizerFast):
+        return tokenizer.tokenize(text, add_special_tokens=True)
+    elif tokenizer.__class__.__name__ == "MockTokenizer":
+        return tokenizer.tokenize(text)
+
+    raise ValueError(f"Unexpected tokenizer: {tokenizer.__class__}")
```

Comment:
What's the reason for preferring this over calling `sentence_transformer.tokenizer.tokenize(text, add_special_tokens=True) directly?

Reply:
`tokenize()` is called in several places and the `add_special_token` argument is dependent on the specified `sentence_transformer.tokenizer`. Rather than remember to update all `tokenize()` calls, we update in one place depending on the tokenizer.